lib/ProductOpener/Display.pm

```diff
@@ -6958,9 +6958,47 @@ sub display_field($$) {
 			}
 		}
 		my $lang_field = lang($field);
-		if ($lang_field eq '') {
+		if ($lang_field eq '' and $field ne 'states') {
 			$lang_field = ucfirst(lang($field . "_p"));
 		}
+		
+		# Separate To-Do and Done Status
+		if ($field eq 'states') {
+			my $done_status = '';
+			my $to_do_status = '';
+			my @status_split = split(',', $value);
+			foreach my $val (@status_split) {
+				if (index($val, "to-be") != -1) {
+		 			$to_do_status .=$val . ",";
+  		 		}
+		 		else {
+		 		$done_status .=$val . ",";
+		 		}
+			}
+			$to_do_status =~ s/,$//;
+			$done_status =~ s/,$//;
+			my $no_entry_to_do_status = $to_do_status;
+			my $no_entry_done_status = $done_status;
+			my $empty = '<a class="tag well_known" href="/state/empty">'."Empty".'</a>';
+			my $incomplete = '<a class="tag well_known" href="/state/to-be-checked">'."To be checked".'</a>';
+			# Regex to extract anchor tag content
+			$empty =~ s/<a[^>]*>|<\/a>//;
+			$incomplete =~ s/<a[^>]*>|<\/a>//;
+			$no_entry_done_status =~ s/<a[^>]*>|<\/a>//;
+			$no_entry_to_do_status =~ s/<a[^>]*>|<\/a>//;
+			if ($no_entry_to_do_status eq $incomplete){
+				$lang_field = "Done";
+				$value = $done_status;
+			}
+			else {
+				$lang_field = "To Do";
+				$value = $to_do_status;
+			}
+			if ($no_entry_done_status ne $empty and $no_entry_to_do_status ne $incomplete){
+				$html .= '<p><span class="field">' . "Done" . separator_before_colon($lc)  . ":</span>" . $done_status . "</p>";
```

Comment:
So here, you can replace "Done" by lang("done_status"), and that will return the translated value for "Done". In English it will be done, but in French it will return "Fait" (once we have translated it of course).

To do that, you need entries like this in the .po file: 

msgctxt "done_status"
msgid "Done"
msgstr "Done"

msgctxt "to_do_status"
msgid "To do"
msgstr "To do"

Reply:
@stephanegigandet, so for entries that are not to be printed, should they not be added in .po files?

Reply:
the .po files are for strings that we want to display on the web site. The web site can be in any language, so any string we display should come from the .po file, thanks to the lang() function.